### PR TITLE
perf: indexvec in group-by

### PR DESCRIPTION
**Pull request by ritchie46**
Try @orlp's indexvec in the group-by operation.

### CODE REVIEW

File: crates/polars-utils/src/idx_vec.rs [hunk 1]
```diff
@@ -127,6 +142,27 @@ impl Drop for IdxVec {
     }
 }
 
+impl Clone for IdxVec {
+    fn clone(&self) -> Self {
```
Comment (orlp) [inline crates/polars-utils/src/idx_vec.rs:146]: This still uses the direct allocator instead of `Vec`.

File: crates/polars-utils/src/idx_vec.rs [hunk 2]
```diff
@@ -148,3 +192,103 @@ impl AsMut<[IdxSize]> for IdxVec {
...
+                new.push(*v)
+            }
+            new
+        } else {
+            value.to_vec().into()
+        }
+    }
+}
+
+impl From<IdxSize> for IdxVec {
```
Comment (orlp) [inline crates/polars-utils/src/idx_vec.rs:251]: I don't like this implementation existing, it is error-prone (and also doesn't exist for `Vec`).

**Review by orlp — CHANGES_REQUESTED** [review]

File: crates/polars-utils/src/idx_vec.rs (hunk 2 above)
Comment (ritchie46) [reply crates/polars-utils/src/idx_vec.rs:251]: Yes, was doubting of this one as well. The goal is a fast new for a single element where we don't do the reserve check.

Maybe `new_from_first`?

File: crates/polars-utils/src/idx_vec.rs (hunk 1 above)
Comment (ritchie46) [reply crates/polars-utils/src/idx_vec.rs:146]: Ah, good point.

**Pull request merged**